src/objects.js

```diff
@@ -1,41 +1,73 @@
 const createPerson = (name, age) => {
-  // your code here
+  const person = {
+	name: name,
+	age: age,
+}
+return person
 };
 
 const getName = object => {
-  // your code here
+  return object.name
 };
 
 const getProperty = (property, object) => {
-  // your code here
+	const personDetails = object 
+  	return personDetails.age  
+
+/*I have no idea how the above can work but {const person = object 
+    return person.property} doesnt work? it should just substitute "age" into property but I get undefined each time?? I'm sure i'm just missing something obvious! */ 
 };
 
 const hasProperty = (property, object) => {
-  // your code here
+  if(object.hasOwnProperty(property) === true){
+	return true}
+else {
+	return false
+}
 };
 
 const isOver65 = person => {
-  // your code here
+const personDetails = person
+    if(personDetails.age > 65) {
+return true
+}
+ else {
+	return false
+}
+
 };
 
 const getAges = people => {
-  // your code here
+return people.map(person => person.age)
 };
 
 const findByName = (name, people) => {
-  // your code here
+  return people.find(person => person.name == name )
 };
 
 const findHondas = cars => {
-  // your code here
+  return cars.filter(maker => maker.manufacturer === "Honda");
 };
 
 const averageAge = people => {
-  // your code here
+ let numberOfPeople = people.length
+ let Ages = people.map(person => person.age)
+ let totalAges = Ages.reduce((a, b) => a + b, 0)
+ let averageAge = totalAges / numberOfPeople
+ return averageAge
+//I feel like this is a long winded way of doing it :/
 };
 
 const createTalkingPerson = (name, age) => {
-  // your code here
+  return {
+	name: name,
+	age: age,
+	Introduce: function greet(newName) {
+	return 'Hi' + newName + ', my name is' + name + 'and I am' + age + '!';
```

Comment:
So, the idea is that when you create a person like:
const khalid = createTalkingPerson('Khalid', 24);

And then call the introduce method:
khalid.introduce('Miguel');

You get something like:

Hi Miguel, my name is Khalid and I am 24!